File: src/main/java/upbrella/be/rent/dto/response/RentalHistoriesPageResponse.java
package upbrella.be.rent.dto.response;

import lombok.Builder;
import lombok.Getter;

import java.util.List;

@Getter
@Builder
public class RentalHistoriesPageResponse {

    private List<RentalHistoryResponse> rentalHistoryResponsePage;
    private long countOfAllHistories;
    private long countOfAllPages;

    public static RentalHistoriesPageResponse of(List<RentalHistoryResponse> rentalHistories, long countOfAllHistories, long countOfAllPages) {

        return RentalHistoriesPageResponse.builder()
                .rentalHistoryResponsePage(rentalHistories)
                .countOfAllHistories(countOfAllHistories)
                .countOfAllPages(countOfAllPages)
                .build();
    }
}

File: src/main/kotlin/upbrella/be/rent/dto/response/RentalHistoriesPageResponse.kt
package upbrella.be.rent.dto.response

data class RentalHistoriesPageResponse(
    val rentalHistoryResponsePage: List<RentalHistoryResponse>,
    val countOfAllHistories: Long,
    val countOfAllPages: Long
) {
    companion object {
        fun of(
            rentalHistories: List<RentalHistoryResponse>,
            countOfAllHistories: Long,
            countOfAllPages: Long
        ): RentalHistoriesPageResponse {
            return RentalHistoriesPageResponse(
                rentalHistoryResponsePage = rentalHistories,
                countOfAllHistories = countOfAllHistories,
                countOfAllPages = countOfAllPages
            )
        }
    }
}
